opencv_ai/splice/io.h

```diff
@@ -0,0 +1,166 @@
+#ifndef OPENCV_AI_SPLICE_IO_H_
+#define OPENCV_AI_SPLICE_IO_H_
+
+#include <opencv_ai/splice/component.h>
+#include <boost/signals2.hpp>
+
+namespace opencv_ai {
+
+template <typename MessageType>
+class ChannelService : public boost::asio::io_service::service {
```

Comment:
To ease review, I stuffed it in one header, I'm kind of indifferent.  I like having a header, translation unit, and unit test, but if it makes sense to group things, for context, I don't mind umbrella headers. They can be split up later.  Do you have a preference?